### PR TITLE
 Configure dbinstance via external Secrets/Configmaps

### PR DESCRIPTION
**Pull request by allanger**
DbInstance now can get a port, a host and a public ip from configmaps/secrets

Also, the dbinstance validation webhook is printing a message that google instances are deprecated

db-operator issue: https://github.com/db-operator/db-operator/pull/102

### CODE REVIEW

File: charts/db-instances/templates/dbinstance.yaml [hunk 1]
```diff
@@ -52,10 +52,22 @@ spec:
   {{- end }}
   {{- if $value.generic }}
   generic:
-    host: {{ $value.generic.host }}
-    port: {{ $value.generic.port }}
+    hostFrom:
+      kind: ConfigMap
+      name: {{ $name }}-config
+      namespace {{ $.Release.Namespace }}
```
Comment (dabde) [inline charts/db-instances/templates/dbinstance.yaml:58]: missing colon `:`

File: charts/db-instances/templates/dbinstance.yaml [hunk 2]
```diff
@@ -52,10 +52,22 @@ spec:
...
+      key: host
+    portFrom:
+      kind: ConfigMap
+      name: {{ $name }}-config
+      namespace {{ $.Release.Namespace }}
```
Comment (dabde) [inline charts/db-instances/templates/dbinstance.yaml:63]: missing colon `:`

File: charts/db-instances/templates/dbinstance.yaml [hunk 3]
```diff
@@ -52,10 +52,22 @@ spec:
...
+      key: port
   {{- if $value.generic.publicIp }}
-    publicIp: {{ $value.generic.publicIp }}
+    publicIpFrom:
+      kind: ConfigMap
+      name: {{ $name }}-config
+      namespace {{ $.Release.Namespace }}
```
Comment (dabde) [inline charts/db-instances/templates/dbinstance.yaml:69]: missing colon `:`

File: charts/db-instances/templates/dbinstance.yaml (hunk 2 above)
Comment (allanger) [reply charts/db-instances/templates/dbinstance.yaml]: should be fixed

File: charts/db-instances/templates/dbinstance.yaml (hunk 3 above)
Comment (allanger) [reply charts/db-instances/templates/dbinstance.yaml]: should be fixed

File: charts/db-instances/templates/dbinstance.yaml (hunk 1 above)
Comment (allanger) [reply charts/db-instances/templates/dbinstance.yaml]: should be fixed

**Pull request merged**